frontend/src/components/Routes/ProtectedRoute.tsx

```diff
@@ -16,22 +15,16 @@ export function ProtectedRoute({
     | React.ComponentType<any>;
   path: string;
 }) {
-  const { userId } = useSelector(
-    (state: AppState) => ({
-      userId: getUserIdFromState(state),
-    }),
-    shallowEqual
+  const userId = useSelector(
+    (state: AppState) => state.studentState?.student?.id
```

Comment:
this fix sort of weirds me out a little bit... previously this was using `getUserIdFromState` which calls `getStudentFromtState(state).id`, which should be fine, but was erroring with a `getStudentFromState is not defined` error, even though it clearly is. however, just inlining the call seems to fix the issue, but still sus.

Reply:
this is very very weird, i haven't seen the getStudentFromState is not defined error yet though. A function just straight up not being defined is very weird 🤔